### PR TITLE
Audio recordings

### PR DESCRIPTION
**Pull request by michaelCasey**
- Users can record tutoring sessions from their upcoming session view
- Recorded sessions can be played from the session history details page

### CODE REVIEW

File: app/src/main/java/cs/dal/krush/seeders/DatabaseSeeder.java [hunk 1]
```diff
@@ -43,7 +43,7 @@ public void insertData() {
         TutoringSessionSeeder.insert(mydb);
         TutorRatingSeeder.insert(mydb);
         CourseSeeder.insert(mydb);
-        AudioRecordingSeeder.insert(mydb);
+//        AudioRecordingSeeder.insert(mydb);
```
Comment (Mons1Oerjan) [inline app/src/main/java/cs/dal/krush/seeders/DatabaseSeeder.java]: should remove these if they are unused

File: app/src/main/java/cs/dal/krush/seeders/DatabaseSeeder.java [hunk 2]
```diff
@@ -59,14 +59,14 @@ public void displayData(){
...
-        rs.moveToFirst();
+//        rs = mydb.audioRecording.getData(1);
+//        rs.moveToFirst();
         String s;
-        s = rs.getString(rs.getColumnIndex("location_id"));
-        System.out.println(s);
-        if (!rs.isClosed()) {
-            rs.close();
-        }
+//        s = rs.getString(rs.getColumnIndex("session_id"));
```
Comment (Mons1Oerjan) [inline app/src/main/java/cs/dal/krush/seeders/DatabaseSeeder.java]: should remove these if they are unused

File: app/src/main/java/cs/dal/krush/studentFragments/StudentUpcSessionsDetailsFragment.java [hunk 3]
```diff
@@ -63,7 +88,10 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
         schoolField = (TextView) view.findViewById(R.id.student_upc_details_school);
         cancelButton = (Button) view.findViewById(R.id.student_cancel_session_button);
         sessionDetailLocation = (Button) view.findViewById(R.id.sessionDetailLocation);
-
+        sessionRecordStart = (Button) view.findViewById(R.id.sessionRecordStart);
+        sessionRecordStop = (Button) view.findViewById(R.id.sessionRecordStop);
+//        sessionRecordStart.setVisibility(view.INVISIBLE);
```
Comment (Mons1Oerjan) [inline app/src/main/java/cs/dal/krush/studentFragments/StudentUpcSessionsDetailsFragment.java]: should remove these if they are unused

File: app/src/main/java/cs/dal/krush/studentFragments/StudentUpcSessionsDetailsFragment.java [hunk 4]
```diff
@@ -106,6 +136,21 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
...
+        DateFormat timeFormatter = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss", Locale.getDefault());
+        Date now = new Date();
+        String currentTime = timeFormatter.format(now);
+        boolean hasStarted = true;
+
+        try {
+            hasStarted = timeFormatter.parse(startTime).before(timeFormatter.parse(currentTime));
+        }
+        catch (Exception e){
+            // TODO Auto-generated catch block
```
Comment (Mons1Oerjan) [inline app/src/main/java/cs/dal/krush/studentFragments/StudentUpcSessionsDetailsFragment.java]: replace comment with ```e.printStackTrace()```

File: app/src/main/java/cs/dal/krush/studentFragments/StudentUpcSessionsDetailsFragment.java [hunk 5]
```diff
@@ -163,8 +208,93 @@ public void onClick(View v) {
...
+                            Environment.getExternalStorageDirectory().getAbsolutePath() + "/" + student.getString(student.getColumnIndex("f_name")) +
+                                    student.getString(student.getColumnIndex("l_name")) + sessionCursor.getString(sessionCursor.getColumnIndex("start_time")) + ".3gp";
+                    Log.e("File name", AudioSavePathInDevice);
+                    MediaRecorderReady();
+
+                    try {
+                        mediaRecorder.prepare();
+                        mediaRecorder.start();
+                    } catch (IllegalStateException e) {
+                        // TODO Auto-generated catch block
```
Comment (Mons1Oerjan) [inline app/src/main/java/cs/dal/krush/studentFragments/StudentUpcSessionsDetailsFragment.java]: remove TODO comment

File: app/src/main/java/cs/dal/krush/studentFragments/StudentUpcSessionsDetailsFragment.java [hunk 6]
```diff
@@ -163,8 +208,93 @@ public void onClick(View v) {
...
+                        e.printStackTrace();
+                    } catch (IOException e) {
+                        // TODO Auto-generated catch block
```
Comment (Mons1Oerjan) [inline app/src/main/java/cs/dal/krush/studentFragments/StudentUpcSessionsDetailsFragment.java]: remove TODO comment

File: app/src/main/java/cs/dal/krush/studentFragments/StudentUpcSessionsDetailsFragment.java [hunk 7]
```diff
@@ -163,8 +208,93 @@ public void onClick(View v) {
...
+                sessionRecordStop.setVisibility(view.INVISIBLE);
+
+            }
+        });
+
         sessionCursor.close();
 
         return view;
     }
+    // init media recorder
```
Comment (Mons1Oerjan) [inline app/src/main/java/cs/dal/krush/studentFragments/StudentUpcSessionsDetailsFragment.java]: Change this comment to javadoc format:
```
/**
 * init media recorder
 */
```

File: app/src/main/java/cs/dal/krush/studentFragments/StudentUpcSessionsDetailsFragment.java [hunk 8]
```diff
@@ -163,8 +208,93 @@ public void onClick(View v) {
...
+    public void MediaRecorderReady(){
+        mediaRecorder=new MediaRecorder();
+        mediaRecorder.setAudioSource(MediaRecorder.AudioSource.MIC);
+        mediaRecorder.setOutputFormat(MediaRecorder.OutputFormat.THREE_GPP);
+        mediaRecorder.setOutputFile(AudioSavePathInDevice);
+        mediaRecorder.setAudioEncoder(MediaRecorder.AudioEncoder.AMR_WB);
+
+    }
+    // check device permissions
```
Comment (Mons1Oerjan) [inline app/src/main/java/cs/dal/krush/studentFragments/StudentUpcSessionsDetailsFragment.java]:
```
/**
 * check device permissions
 */
```

File: app/src/main/java/cs/dal/krush/studentFragments/StudentUpcSessionsDetailsFragment.java [hunk 9]
```diff
@@ -163,8 +208,93 @@ public void onClick(View v) {
...
+    public boolean checkPermission() {
+        int result = ContextCompat.checkSelfPermission(getContext(),
+                WRITE_EXTERNAL_STORAGE);
+        int result1 = ContextCompat.checkSelfPermission(getContext(),
+                RECORD_AUDIO);
+        return result == PackageManager.PERMISSION_GRANTED &&
+                result1 == PackageManager.PERMISSION_GRANTED;
+    }
+    // set device permissions
```
Comment (Mons1Oerjan) [inline app/src/main/java/cs/dal/krush/studentFragments/StudentUpcSessionsDetailsFragment.java]:
```
/**
 * set device permissions
 */
```

File: app/src/main/java/cs/dal/krush/studentFragments/StudentUpcSessionsDetailsFragment.java (hunk 3 above)
Comment (michaelCasey) [inline app/src/main/java/cs/dal/krush/studentFragments/StudentUpcSessionsDetailsFragment.java]: This is here if we get the dates to function accordingly

**Comment by Mons1Oerjan** [comment]
Approved!

**Pull request merged**